back-nest/src/ws/ws.gateway.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  WebSocketGateway,
+  WebSocketServer,
+  SubscribeMessage,
+  MessageBody,
+} from '@nestjs/websockets';
+import { Logger } from '@nestjs/common';
+import { Server } from 'socket.io';
+import Redis from 'ioredis';
+import Redlock, { ExecutionError } from 'redlock';
+
+import {
+  ScraperServiceFactory,
+  ServiceType,
+} from '../sсrapers/ScraperServiceFactory';
+import { PrismaService } from '../../prisma/prisma.service';
+
+enum SourceType {
+  telemart = 'TELEMART',
+  rozetka = 'ROZETKA',
+}
+
+@WebSocketGateway({ cors: true })
+export class ScraperGateway {
+  @WebSocketServer() server: Server;
+  private readonly logger = new Logger(ScraperGateway.name);
+  private readonly redisClient: Redis;
+  private readonly redlock: Redlock;
+
+  constructor(
+    private readonly scraperServiceFactory: ScraperServiceFactory,
+    private readonly prisma: PrismaService,
+  ) {
+    const redis = new Redis({
+      host: 'redis',
+      port: 6379,
+    });
+    this.redisClient = redis;
```

Comment:
You already have redis in redis.service. Why are you initializing it twice in different places?